public/app/features/plugins/extensions/errors.ts

```diff
@@ -0,0 +1,42 @@
+export const INVALID_EXTENSION_POINT_ID =
+  'Invalid extension point. Reason: Extension point id should be prefixed with your plugin id, e.g "myorg-foo-app/toolbar/v1".';
+
+export const EXTENSION_POINT_META_INFO_MISSING =
+  'Invalid extension point. Reason: The extension point is not recorded in the "plugin.json" file. Extension points must be listed in the section "extensions.extensionPoints[]". Returning an empty array of extensions.';
+
+export const TITLE_MISSING = 'Title is missing.';
+
+export const DESCRIPTION_MISSING = 'Description is missing.';
+
+export const INVALID_CONFIGURE_FUNCTION = 'The "configure" function is invalid. It should be a function.';
+
+export const INVALID_PATH_OR_ON_CLICK = 'Either "path" or "onClick" is required.';
+
+export const INVALID_PATH = 'The "path" is required and should start with "/a/<pluginId>".';
+
+export const INVALID_EXPOSED_COMPONENT_ID =
+  "The component id does not match the id naming convention. Id should be prefixed with plugin id. e.g 'myorg-basic-app/my-component-id/v1'.";
```

Comment:
I am wondering if we should prefix these logic messages with something like `Invalid exposed component.`, similar to the invalid extension point messages?

Reply:
It's not obvious by looking at the errors here, but they are actually already prefixed (see example [here](https://github.com/grafana/grafana/blob/extensions/streamline-log-messages/public/app/features/plugins/extensions/registry/AddedLinksRegistry.ts#L53)). I'll make this a func instead that takes the prefix as an arg so that it becomes it's easier to read all the error messages in this file. Thanks!

Reply:
Hmm changing my mind. Think it's better to have the error messages pure without prefixes in this file. That way we can better control the output if we ever want to validate the entire config before halting registration.